setup.py

```diff
@@ -15,6 +15,27 @@
 exec(compile(open(mod).read(), mod, 'exec'), {}, pkg)
 
 
+def parse_requirements(requirements, ignore=('setuptools',)):
+    """Read dependencies from requirements file and strip off version numbers.
+
+    Note: this imlementation does not support requirements files with extra
+    requirements (in square brackets) or '>=' and '<=' version qualifiers.
+
+    """
+    with open(requirements) as f:
+        packages = set()
+        for line in f:
+            line = line.strip()
+            if line.startswith(('#', '-r', '--')):
+                continue
+            if '#egg=' in line:
+                line = line.split('#egg=')[1]
+            pkg = line.split('==')[0].strip()
```

Comment:
Yes we can, but we cannot really predict the future. For an application which will probably share an env with lots of other others apps, it's better to be inclusive IMO.